docs/index.rst

```diff
@@ -17,6 +21,32 @@ More in-depth coverage is provided in other sections.
     api/index
     examples
 
+
+Concepts
+--------
+
+Besides code which implements Widgetastic and navmazing functionality, two
+basic concepts used to abstract Satellite/Foreman UI functionality are Views
+and Entities.
+
+**View** is part of page containing widgets, it describes appearance of the
+page. Here all the nasty XPaths goes, but please create Foreman or Katello
+pull requests for `id=...` additions where reasonable, as generally XPaths
+are too fragile.
+
+**Entity** is a functional side of application (Satellite or Foreman in our
```

Comment:
**Entity** is fundamental basic concept that is responsible for functional side of application (Satellite in our case). It defines actions you can do with objects like CRUD (create, update, delete) for example. That is only API that is visible for end user on test side. Also, we put all details about navigation here, so framework know how to get to necessary page when you need to create specific object in the application